tests/library_wrappers/test_sparsehash.cc

```diff
@@ -42,4 +43,17 @@ TEST(dense_hash_set, invalid_empty_key) {
     EXPECT_THROW((M8_key{py::datetime64ns::nat()}), std::invalid_argument);
     EXPECT_THROW((M8_key{py::datetime64ns::nat(), 10}), std::invalid_argument);
 }
+
+TEST(any_vtable, map_key) {
```

Comment:
Can you move this test back to `test_any.cc` and just use a `std::unordered_map`?
The point of this test is to ensure that `any_vtable` can be used as a map key, which should be true even if you don't use sparsehash.

Reply:
whoops - fixed